src/app/teacher-session.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import { FirebaseService } from './firebase.service';
+import { SessionStudentData } from './session';
+import { Observable, BehaviorSubject, of } from 'rxjs';
+import { switchMap, shareReplay } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TeacherSessionService {
+
+  constructor(private firebaseService: FirebaseService) { }
+
+  sessionId$ = new BehaviorSubject<string | null>(null);
+
+  /**
+   * This observable lists the students in the session that the teacher is currently connected to.
+   * (If the teacher isn't connected to a session, then it emits null.)
+   *
+   * This observable emits under the following circumstances:
+   * - when you initially subscribe to it,
+   * - when the teacher joins or leaves a session,
+   * - and when the contents of the current session change in Firebase.
+   */
+  studentsInCurrentSession$: Observable<SessionStudentData[] | null> = this.sessionId$.pipe(
+    switchMap(sessionId =>
+      sessionId
+        ? this.firebaseService.getStudentsInSession(sessionId)
+        : of(null)
```

Comment:
It would probably be easier to consume this if it gave an empty array when there is no session. This is consistent with how we do array observables on the student side.